plugins/history/chathistoryhandler.h

```diff
@@ -0,0 +1,60 @@
+#ifndef CHATHISTORYHANDLER_H
+#define CHATHISTORYHANDLER_H
+
+#include "databasemanager.h"
+#include <QObject>
+
+class QDate;
+
+/**
+ * The ChatHistoryHandler class is used to handle all history connected activities, and
+ * then call the respective class (logger, searcher etc).
+ */
+class ChatHistoryHandler : public QObject
```

Comment:
Isn't `ChatHistoryHandler` doing the same thing as `HistoryPlugin` ? or rather Will it be a good idea to merge these two classes?